app-web/src/utils/contentContribution.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * takes in a 'source' object from the TopicForm and reduces it to the
+ * expected format from the registry
+ * @param {Object} source the source (could be web or github)
+ * @param {String} source.sourceType
+ * @param {Object} source.sourceProperties
+ * @param {String} operation the form mode (edit or create)
+ */
+export const reduceFormSourceToSourceProperties = ({ sourceType, sourceProperties }) => {
+  if (sourceType === 'github') {
+    return {
+      ...sourceProperties,
+      files: sourceProperties.files.split(',').map(file => file.trim()),
```

Comment:
This will still not work for the edit topic feat. The file data is coming in as an array. You can't perform .trim and .split on an array. Needs to be converted to a string first.

Reply:
im still really confused on how the implementations change between edit and create